### DISCUSSION TITLE
Plot should be able to show the dot size scale in a legend

### DISCUSSION THREAD

**Discussion by llimllib** [observablehq/plot#1106] [Q&A] [answered]
As far as I can tell, if you use the `r` scale in a dot plot, there is no way to give the reader of the plot any indication of what the radius represents.

- Is that accurate? I don't see any examples of a legend for dot size
- If so, plot should have a way to easily tell the reader what is being represented with

Having the `r` channel available by default, but not label-able by default (if that is indeed the case!) is likely to make it confusing unless the chart's author goes well out of their way to provide a label.

I made a simple [demonstration of what I mean here](https://observablehq.com/d/630306124fd47ced)

## How should it look?

I don't honestly know, this is an area for research! A graph tool I wrote in d3 uses a circle mark:

<img width="408" alt="image" src="https://user-images.githubusercontent.com/7150/201223142-592da7bf-a988-425a-986a-3f6e59d54515.png">

But honestly I'm not super happy with how well that conveys the information, and many people here are vastly better at visual design than myself. I suppose I should go take a look in the semiology of graphics for ideas, but I'm not at home so I can't look right now.

**Comment by llimllib** [comment]
Vega-lite [puts a legend to the side](https://vega.github.io/vega-lite/examples/circle_binned.html)

<img width="468" alt="image" src="https://user-images.githubusercontent.com/7150/201223958-704c3a1d-8b5d-4af4-adb5-00c2abdcf71a.png">

**Comment by llimllib** [comment]
ggplot [also puts a legend to the side](https://ggplot2.tidyverse.org/reference/geom_point.html)

<img width="640" alt="image" src="https://user-images.githubusercontent.com/7150/201224770-0a915bf9-e8a9-48a8-8c1c-a1553ef95f59.png">

**Comment by llimllib** [comment, marked as answer]
Oh! It appears to be [considered but stalled](https://github.com/observablehq/plot/pull/665). Consider this a request for the feature. I don't as much care what exactly the legend looks like as that I can convey to my viewers what the values mean at all!

**Reply by Fil** [reply]
Yes it's a bit stalled, thanks for voting it up :)

In the meantime a solution is to name your chart, then use `chart.scale("r")` to draw your own, with Plot, D3, or even in raw SVG.

**Reply by llimllib** [reply]
I don't have any practical idea what that means, but I'll cross that bridge when I get to it. Thanks for the update

**Reply by Fil** [reply]
This notebook shows a way to use the code from the PR:
https://observablehq.com/@recifs/a-radius-legend-for-plot-665
